MediaManager/MediaManagerBase.cs

```diff
@@ -60,14 +60,46 @@ public virtual IMediaQueue MediaQueue
         public abstract RepeatMode RepeatMode { get; set; }
 
         public abstract Task Pause();
-        public abstract Task Play(IMediaItem mediaItem);
-        public abstract Task<IMediaItem> Play(string uri);
-        public abstract Task Play(IEnumerable<IMediaItem> items);
-        public abstract Task<IEnumerable<IMediaItem>> Play(IEnumerable<string> items);
+        public virtual Task Play(IMediaItem mediaItem)
+        {
+            MediaQueue.Clear();
+            MediaQueue.Add(mediaItem);
+            return Task.CompletedTask;
+        }
+        public async virtual Task<IMediaItem> Play(string uri)
+        {
+            var mediaItem = await MediaExtractor.CreateMediaItem(uri);
+            MediaQueue.Clear();
+            MediaQueue.Add(mediaItem);
+
+            return mediaItem;
+        }
+        public virtual Task Play(IEnumerable<IMediaItem> items)
+        {
+            MediaQueue.Clear();
+            foreach (var item in items)
+            {
+                MediaQueue.Add(item);
+            }
+
+            return Task.CompletedTask;
+        }
+        public async virtual Task<IEnumerable<IMediaItem>> Play(IEnumerable<string> items)
+        {
+            MediaQueue.Clear();
+
+            foreach (var uri in items)
+            {
+                var mediaItem = await MediaExtractor.CreateMediaItem(uri);
+                MediaQueue.Add(mediaItem);
+            }
+
+            return MediaQueue;
+        }
         public abstract Task<IMediaItem> Play(FileInfo file);
         public abstract Task<IEnumerable<IMediaItem>> Play(DirectoryInfo directoryInfo);
         public abstract Task Play();
-        public abstract Task PlayNext();
+        public abstract Task<bool> PlayNext();
```

Comment:
It would make sense to add a bool to both PlayNext and PlayPrevious and when there is none return false?

Reply:
Done!